src/utils/FillUtils.ts

```diff
@@ -0,0 +1,33 @@
+import { RISK_LABS_RELAYER_ADDRESS } from "../constants";
+import { Deposit, Fill } from "../interfaces";
+import { bnZero } from "./BigNumberUtils";
+
+export function buildFillForSimulatingFullDeposit(deposit: Deposit, relayerAddress = RISK_LABS_RELAYER_ADDRESS): Fill {
+  return {
+    amount: deposit.amount,
+    fillAmount: deposit.amount, // We're simulating a full fill
```

Comment:
So I'm sure you're considering how to handle this in a future PR but there is a chance that the `deposit.amount > simulated relayer token balance` leading to unexpected reverts FYI

Reply:
We handle it in a later case: 
```
  if (!isMessageEmpty) {
    const isRecipientAnAddress = ethers.utils.isAddress(recipientAddress);
    const isRelayerAnAddress = ethers.utils.isAddress(relayerAddress);
    if (!isRecipientAnAddress || !isRelayerAnAddress) {
      throw new Error("Could not simulate message fill. Recipient address or relayer address is not a valid address");
    }
    const [isRecipientAContract, relayerBalanceOfToken] = await Promise.all([
      isContractAddress(recipientAddress, provider),
      getTokenBalance(relayerAddress, fillToSimulate.destinationToken, provider),
    ]);
    if (!isRecipientAContract) {
      throw new Error("Could not simulate message fill. Recipient address is not a contract address");
    }
    if (toBN(relayerBalanceOfToken).lt(toBN(amountToRelay))) {
      throw new Error("Could not simulate message fill. Partial fills are not supported with message relaying");
    }
  }
```

Reply:
If `toBN(relayerBalanceOfToken).lt(toBN(amountToRelay))` is true, then we unwind the stack and don't actually simulate the costs.